sdk/inc/azure/storage/az_storage_blobs.h

```diff
@@ -108,38 +108,38 @@ typedef struct
 AZ_NODISCARD az_storage_blobs_blob_client_options az_storage_blobs_blob_client_options_default();
 
 /**
- * @brief Gets the default blob upload options
+ * @brief Gets the default blob upload options.
  *
- * @details Call this to obtain an initialized #az_storage_blobs_blob_upload_options structure
+ * @details Call this to obtain an initialized #az_storage_blobs_blob_upload_options structure.
  *
  * @remark Use this, for instance, when only caring about setting one option by calling this method
  * and then overriding that specific option.
- *
  */
 AZ_NODISCARD AZ_INLINE az_storage_blobs_blob_upload_options
 az_storage_blobs_blob_upload_options_default()
 {
   return (az_storage_blobs_blob_upload_options){ .context = &az_context_application,
-                                                ._internal = { .unused = AZ_SPAN_NULL } };
+                                                 ._internal = { .unused = AZ_SPAN_NULL } };
 }
 
 /**
  * @brief Uploads the contents to blob storage.
  *
- * @param client A storage blobs client structure.
- * @param content The blob content to upload.
- * @param options __[nullable]__ A reference to an #az_storage_blobs_blob_upload_options
+ * @param client An #az_storage_blobs_blob_client structure.
```

Comment:
[in]

Reply:
I was about to say that the client shouldn't be marked with `[in]`.

But, for some reason, GitHub is confusing what line this comment corresponds to.
![image](https://user-images.githubusercontent.com/6527137/90440266-c9446480-e08b-11ea-932c-120ef670ec09.png)

In the "Files Changed" view, you can see this comment is about the credential parameter, not the client.
https://github.com/Azure/azure-sdk-for-c/pull/1057/files
![image](https://user-images.githubusercontent.com/6527137/90440301-db260780-e08b-11ea-8ee5-fdd7f37da352.png)

Reply:
The reason I didn't mark it is that logically it does not mutate, but factually it does. But since all the mutations are in the `_internal`, these can be marked as `[in]` I think.

Reply:
I would think about what contract we want to document to the customer and what expectations we want them to have, especially as it changes in the future. If we are mutating the internal, and that is necessary, can we decide to make those fields public and not go against the "documented contract"?

If we want the flexibility, and do mutate it today, I would lean towards not marking it as `[in]` to leave that option open. If we don't ever foresee mutating it in the future, and we want to give that guarantee to the user, only then should you mark it as `[in`].

Reply:
>  but factually it does.

What mutations are we doing to the client on a call to upload (even internal state)? That seems surprising to me.